src/planscape/plan/tests.py

```diff
@@ -718,6 +718,50 @@ def test_get_project_with_priorities(self):
                          self.condition1.pk, self.condition2.pk])
 
 
+class ListProjectsTest(TransactionTestCase):
+    def setUp(self):
+        self.user = User.objects.create(username='testuser')
+        self.user.set_password('12345')
+        self.user.save()
+
+        self.geometry = {'type': 'MultiPolygon',
+                         'coordinates': [[[[1, 2], [2, 3], [3, 4], [1, 2]]]]}
+        stored_geometry = GEOSGeometry(json.dumps(self.geometry))
+
+        self.plan_with_user = create_plan(
+            self.user, 'plan', stored_geometry, [])
+
+    def test_list_no_plan_id(self):
+        self.client.force_login(self.user)
+        response = self.client.get(
+            reverse('plan:list_projects_for_plan'),
+            {},
+            content_type="application/json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_list_no_projects(self):
```

Comment:
nit: I can see this happening if a plan doesn't exist or if a plan has no projects. If a plan doesn't exist or if the user can't see the plan, would it be useful to return an error message instead?

Reply:
test_list_no_plan_id somewhat covers (1) I can add a test for (2)

We return an error message with the error code and the message is generated by the django model framework i.e. something like Project for this id is not found. did you want to annotate the message?

Reply:
It'd be cool to see annotated error messages - but I also think it's more of a TODO than a blocker depending on FE needs.

Reply:
sg!